### DISCUSSION TITLE
Representing outcome data

### DISCUSSION THREAD

**Discussion by philipmjohnson** [agilegardenclub/public-garden#47] [Ideas]
## Motivation

One of the primary ways that gardeners can gain insight from each other is by sharing outcome data about plantings: did the planting survive to harvest, and if so, how was the appearance of the food, the flavor, etc?

The goal of this discussion is to decide upon a standard representation for outcome data that is simple enough for gardeners to use in a consistent manner, and sophisticated enough for gardeners to improve their decision making when planning their next garden.

**Status as of August, 2022: We are moving forward with the representation described below.**

## The outcome scale

We currently represent outcomes using a star system. For each outcome category, a planting can be rated from one to five.  You cannot give "zero stars" as an outcome rating, because "zero stars" represents the absence of an outcome rating. 

## Outcome Categories and Ratings

Note that all of these rankings are designed so that "1" is the worst and "5" is the best. 

### Germination

Germination enables gardeners to represent what percentage of seeds germinated.

| Rating | Definition |
| ------  | ----------|
| 1 |  **Failure.**  No seeds germinated. |
| 2 |  **Poor.** Approximately a quarter of the seeds germinated. |
| 3 |   **OK.** Approximately half of the seeds germinated. |
| 4 |   **Good.** Approximately 3/4 of the seeds germinated. |
| 5 |   **Outstanding.** 90% or more of the seeds germinated. |

### Yield

Yield refers to the amount of food produced by a planting.  This is somewhat subjective, but can still provide guidance to others about the success of a varietal when yield values are aggregated over a chapter. 

| Rating | Definition |
| ------  | ----------|
| 1 |   **None.** The planting died and/or did not yield any food. |
| 2 |   **Minimal.**  The planting yielded significantly less food than expected. |
| 3 |   **OK.** The planting yielded the expected amount of food. |
| 4 |   **Good.** The planting yielded somewhat more food than expected. |
| 5 |   **Outstanding.**  The planting yielded significantly more food than expected. |

### Flavor

Flavor ranges from bad to great. 

| Rating | Definition |
| ------  | ----------|
| 1 |   **Bad.** Not worth eating. |
| 2 |   **Bland.** Worth eating, but only a little. |
| 3 |   **OK.**  Expected level of flavor.  |
| 4 |   **Good.** Better than OK flavor, enjoyable to eat.|
| 5 |   **Outstanding.** Can't imagine it tasting better. |

### Pest and disease resistance

Pest and disease resistance is defined as the percentage of plantings (either the plant or the harvested crop) that suffer damage from pests and/or disease.

| Rating | Definition |
| ------  | ----------|
| 1 |  **Extremely poor.** 90% or more of the plantings have damage. |
| 2 |  **Very poor.** More than half of the plantings have damage. |
| 3 |  **OK.**  No more than a quarter of plantings have damage. |
| 4 |  **Good.** Only a few plantings have damage.  |
| 5 |  **Outstanding.** No observable damage.  |

### Appearance 

Appearance refers to the harvested crop, not the plant.  If we care about the appearance of the plant, we need to create a separate outcome category so we can keep these two categories of appearance separate.

Appearance refers to the shape, shape, and coloration of the harvested crop. The rating system judges the percentage of the crop exhibiting appearance problems. 

We call a harvested crop "ugly" if it suffers from significant shape, size, and/or coloration problems. We call it "normal" if it has small shape, size, and/or coloration problems.  We call it "beautiful" if it has almost no shape, size, or coloration problems.

| Rating | Definition |
| ------  | ----------|
| 1 |   **Almost all ugly.** 90% or more of the crop is ugly. |
| 2 |   **Mostly ugly.** Over 50% of the crop is ugly. |
| 3 |   **Mostly OK.** Over 50% of the crop is OK.  |
| 4 |   **Mostly beautiful.** Over 50% of the crop is beautiful. |
| 5 |   **Almost all beautiful.** 90% or more of the crop is beautiful. |

**Comment by Jennacd** [comment]
I've been struggling a bit with the concept of outcome data because of how subjective it is to the gardener, their soil, watering habits, wildlife, experience etc.  At the same time, I do think we should provide a simple way for gardeners to keep track of their plant outcomes, and I do think that there is value to share this data on a chapter level.  When I remind myself about why we want to include outcome data it comes down to three motivations.  It allows gardeners a simple way to:
A) Remember how well their plants produced in years past, so they can decide whether or not to grow that crop again
B) Shop through other crops being grown in their area and make an informed decision on if they want to try a new crop out
C) Provide a quick, low effort way to track/share crop success levels

I propose that we take out germination, pest resistance, and appearance and only ask for outcome data on flavor and yield.  These are the two most important variables to consider when growing food, besides "can I grow it in my area."  By reducing outcomes measures to two, my theory is that more gardeners will actually take the time to enter this data.

The issue with germination outcome data is that outcome data probably reflects the gardener more than the seeds.  Seeds are tested for germination success and some seed packets will even list this.  So when less sprout, it's relatively safe to assume that you have something to do with it - bad sowing technique, timing, wrong temperature, soil composition, moisture levels or a mix of these things.

I run up against this with pest resistance too - gardeners will have different levels of aggression against pests, so do the ratings assume zero interference, some, or an aggressive approach?  And then there are so many ways to deal with pests - some plants may bounce back well after certain forms of intervention but not for all forms of intervention.  I can't get around how to make this less subjective.

Appearance...just not as important as taste and yield, when it comes to growing food.  Also I wonder, if something tastes good, are we more likely to think it looks good?  And if it looks good are we more likely to think it tastes good?  It is said that "digestion starts in the eyes."

If we stick to flavor and yield, gardeners can quickly rate crops on what (imo) matters most.  If they want to say more, that's great, that's why we have observations.

Thoughts?

**Reply by philipmjohnson** [reply]
Hmm.  While I agree that outcome data should be "quick" and "low effort", I don't think that the solution is "only make gardeners provide two measures".  This is because (1) it's possible that in some situations, germination data (for example) might actually be important---for example, when folks are using shared seeds and want to get some field data because there's no seed packet, and (2) for some gardeners, even providing those two outcome measures might be "too much effort". 

Alternatively, I suggest we provide all the possible outcome measures we think could be significantly useful, but make it clear that providing data on any of them is optional, and design the UI so that they don't get in the way if not supplied. 

The subjectivity problem is important. If there's no way to consistently define an outcome value, then it's not useful.  I guess we could think about letting gardeners define the outcome measures themselves, but then they aren't sharable, which is bad for the chapter. 

So, if you don't think we can come up with good ways to "measure" germination, appearance, and pest resistance, then I'd be open to discarding them for now. But I think that should be the criteria for inclusion/exclusion of an outcome measure, not the overhead involved. 

For me, it seems like we're asking them to incur a LOT of data entry "overhead" when they use AGC.  In the scheme of things, adding a few outcome measures (if we can define them appropriately) doesn't seem like a big deal.

**Reply by Jennacd** [reply]
I'm not concerned as much with the gardener's subjectivity, I'm more so concerned with the other variables.  Germination for example - Take two gardeners - they both sow seeds and one doesn't water them regularly and the other does.  Their germination rates don't reflect the seeds, they reflect the gardeners.  We could decide to assume that people using the app are "good enough" gardeners that any mistakes seed sowing will be a small fraction in comparison to outcomes from gardeners sowing seeds/watering etc correctly.  I can live with that for germination.

Pest **and disease** resistance (Let's add disease in):  So I've been toying with a new rating system that includes levels of intervention, but it keeps becoming too complicated. I've settled with adding "with or without intervention" into the description so that it is explicit.

Appearance of harvest: I am going to keep pushing back on this one.  I think ultimately it will be duplicative of pest/disease resistance.  If we decide to include it we need to define "normal" - would that be how it would look in a grocery store?  And so beautiful is anything better than that?

I'll try to chat with some gardener friends tomorrow and see what they think.

**Comment by Jennacd** [comment]
As for the proposed ratings scale:  I'm ok with what is proposed for Yield and Flavor.  Yield makes perfect sense to me.  Flavor did give me a bit of pause, because I often find the flavor to be pretty uniform thoughout, not 50% good and 50% bad, but I can see how that sometimes could be the case.  When I think about rating crops that taste uniform throughout, I come up with something like this:
1 - Does not taste good, not worth eating
2 - Tastes just ok, worth eating a little.
3 - Tastes average, ok to eat.
4 - Tastes good, enjoyable to eat.
5 - Tastes great, can't imagine it tasting better.

**Comment by Jennacd** [comment]
I've chatted with a few gardeners.  Here was their feedback:

(Jessie) _I would be most interest in taste and yields. Though i do consider disease resistence especially in cucumbers and tomatoes._  

(Jess and Rory) _As far as germination and pest/disease resistance go, they’re important factors to consider and while there will be some shitty gardeners who don’t know what they’re talking about, it’ll all even out. Like, if 50% of your gardeners are shitty gardeners and they can’t get a variety of carrot to germinate but your 50% of decent gardeners can, well that carrot gets a 50% germination rating. Then some carrots that can take a lot of abuse and even shitty gardeners can get them to pop up, their germination rating goes higher. I think since your audience will be a mixed bag that this is probably okay. I hope that makes sense.  And we both think that appearance is important. It’s totally subjective but there are people out there who just love a pretty head of lettuce 🤷‍♀️ and I get it. Not the most valuable, but don’t count it out._ 

Jess and Rory continue to add: _And the other factors are all good. If there was one thing missing, we both think some rating about timing (how long the plant is in the ground) could have a rating. Not that time in the ground is good or bad, but if something is rated as fast to grow I think novice growers would want to know that. Event though it says it on the seed packet, those days to maturity aren’t always accurate for our area (especially if you live in a shadier yard or buy non-local seed). Not sure what the rating would be called but just an idea._ 

Which is exactly what we'll be able to provide from gardener's planning and actual garden timeline data!  

So, I think we keep in all outcome data and I keep working on defining the ratings!

**Comment by philipmjohnson** [comment]
I've just done an editing pass over the outcome definitions. 

(1) I've incorporated Jenna's revised specs for flavor. I think she's right that flavor is probably more-or-less uniform.

(2) I've removed the use of the term "average", because that's a statistical term and would be hard for folks to determine. Instead, I've started using "OK", which I think is closer to the "vibe of the thing".

**Comment by philipmjohnson** [comment]
I have done a small edit on the original posting to clarify that this is the outcome representation we will use for now.
